### PR TITLE
Task: 04-arrays-tasks

### PR DESCRIPTION
**Pull request by verenichyi**
Implement `findElement`, `generateOdds`, `doubleArray`, `getArrayOfPositives`, `getArrayOfStrings`, `removeFalsyValues`, `getUpperCaseStrings`, `getStringsLength`, `insertItem`, `getHead`, `getTail`, `toCsvText`, `toArrayOfSquares`, `getMovingSum`, `getSecondItems`, `propagateItemsByPositionIndex`, `get3TopItems`, `getPositivesCount`, `sortDigitNamesByNumericOrder`, `getItemsSum`, `getFalsyValuesCount`, `findAllOccurences`, `toStringList`, `sortCitiesArray`, `getIdentityMatrix`, `getIntervalArray`, `distinct`, `group`, `selectMany`, `getElementByIndexes`, `swapHeadAndTail` funcs

### CODE REVIEW

File: js/practice/task/04-arrays-tasks.js [hunk 1]
```diff
@@ -321,81 +339,106 @@ function getPositivesCount(arr) {
...
+		three: 3,
+		four: 4,
+		five: 5,
+		six: 6,
+		seven: 7,
+		eight: 8,
+		nine: 9,
+	};
+
+	return arr.sort((a, b) => map[a] - map[b]);
```
Comment (echonok) [inline js/practice/task/04-arrays-tasks.js]: I would only add toLowerCase

File: js/practice/task/04-arrays-tasks.js [hunk 2]
```diff
@@ -529,70 +595,71 @@ function selectMany(arr, childrenSelector) {
...
+ *   [ 1, 2, 3, 4, 5, 6, 7, 8 ]   =>  [ 5, 6, 7, 8, 1, 2, 3, 4 ]
  *
- *   [ 1, 2 ]  => [ 2, 1 ] 
- *   [ 1, 2, 3, 4, 5, 6, 7, 8 ]   =>  [ 5, 6, 7, 8, 1, 2, 3, 4 ]   
- * 
  */
 function swapHeadAndTail(arr) {
-    throw new Error('Not implemented');
+	const mid = Math.ceil(arr.length / 2);
+	return arr.map((x, i) => (i < mid) ? arr[mid + i] || x : arr[i - mid]);
```
Comment (echonok) [inline js/practice/task/04-arrays-tasks.js]: Look, your solution does the trick but consumes too much effort. We don't deen to loop through an array. Why don't use `slice`?

**Review by echonok — CHANGES_REQUESTED** [review]

File: js/practice/task/04-arrays-tasks.js (hunk 1 above)
Comment (verenichyi) [reply js/practice/task/04-arrays-tasks.js]: Refactored

File: js/practice/task/04-arrays-tasks.js (hunk 2 above)
Comment (verenichyi) [reply js/practice/task/04-arrays-tasks.js]: Refactored

**Review by echonok — APPROVED** [review]

**Pull request merged**